packages/pages/src/upgrade/migrate_config.ts

```diff
@@ -0,0 +1,145 @@
+};
+
+const migrateCiJson = async (configYamlPath: string, ciPath: string) => {
+  const ciJson = readJsonSync(ciPath);
+  if (ciJson !== null) {
+    ciJson.artifactStructure = undefined;
+    console.info("migrating site config from ci.json to config.yaml");
+    writeYamlSync(configYamlPath, "ciConfig", ciJson);
+  }
+};
+
+const migrateLocales = async (configYamlPath: string, featuresPath: string) => {
+  const featuresJson = readJsonSync(featuresPath);
+  if (!!featuresJson && !!featuresJson.locales) {
+    console.info("migrating locales from features.json to config.yaml");
+    writeYamlSync(configYamlPath, "locales", featuresJson.locales);
+  }
+};
+
+const migrateSiteStream = async (
+  configYamlPath: string,
+  siteStreamPath: string
+) => {
+  const sitesJson = readJsonSync(siteStreamPath);
+  if (sitesJson !== null) {
+    console.info("migrating global data from site-stream.json to config.yaml");
+    writeYamlSync(configYamlPath, "globalData", sitesJson);
+  }
+};
+
+const migrateRedirects = async (
+  configYamlPath: string,
+  redirectsPath: string
+) => {
+  if (fs.existsSync(redirectsPath)) {
+    console.info("migrating redirects from redirects.csv to config.yaml");
+    const lines = fs.readFileSync(redirectsPath, "utf-8").split("\n");
+    const redirects: any[] = [];
+    lines.forEach((line: string) => {
+      const s: string[] = line.split(",");
+      redirects.push({
+        from: s[0],
+        to: s[1],
+      });
+    });
+    writeYamlSync(configYamlPath, "redirects", redirects);
+  }
+};
+
+const migrateServing = async (configYamlPath: string, servingPath: string) => {
+  const servingJson = readJsonSync(servingPath);
+  if (!!servingJson && !!servingJson.displayUrlPrefix) {
+    console.info(
+      "migrating reverse proxy info from serving.json to config.yaml"
+    );
+    writeYamlSync(configYamlPath, "reverseProxy", {
+      displayUrlPrefix: servingJson.displayUrlPrefix,
+    });
+  }
+};
+
+const migrateSiteMap = async (configYamlPath: string, sitemapPath: string) => {
+  const sitemapJson = readJsonSync(sitemapPath);
+  if (sitemapJson !== null) {
+    console.info("migrating sitemap from sitemap.json to config.yaml");
+    writeYamlSync(configYamlPath, "sitemap", sitemapJson);
+  }
+};
+
+const migrateAuth = async (configYamlPath: string, authPath: string) => {
+  const authJson = readJsonSync(authPath);
+  if (authJson !== null) {
+    console.info("migrating auth info from auth.json to config.yaml");
+    writeYamlSync(configYamlPath, "authorization", authJson);
+  }
+};
+
+/**
+ * Migrates configuration files from sites-config to config.yaml
+ * @param projectStructure the structure of the project
+ */
+export const migrateConfigs = async (projectStructure: ProjectStructure) => {
+  const sitesConfigPath = projectStructure
+    .getSitesConfigPath()
+    .getAbsolutePath();
+  const configYamlPath = path.resolve(projectStructure.config.rootFiles.config);
```

Comment:
We'll need to incorporate the scope when creating the config.yaml. I had a meeting today and we decided that we will use folder scopes. sites-config/ is already set up like this so we just need to move the things in the same way.

```
brandone.com/
    config.yaml
brandtwo.com/
   config.yaml
config.yaml -- this is non-scoped 
```

Reply:
If there a multiple scopes like so:
sites-config:
    ..configStuff
    tacos: ..configStuff

And a user runs npx pages upgrade without a scope, should tacos get deleted? The user may still need to migrate it.

Or should upgrade not accept a scope and simply migrate everything?

I think the latter

Reply:
Good point. We should just do everything during the one-time upgrade command and move everything, including all scopes.

Reply:
Done both

If you give it a scope, it migrates everything from that scope and down and doesn't delete sites-config for the unmigrated scopes.

If you don't provide a scope (or just root), it simply migrates everything.